packages/react-app/components/notifications/OnboardingBatch.jsx

```diff
@@ -80,55 +100,62 @@ const OnboardingBatch = ({ notification, onMarkAsRead, builder, userProvider, on
   };
 
   const handleJoinTelegram = () => {
-    window.open(NEXT_BATCH_TELEGRAM);
+    window.open(latestOpenBatch?.telegramLink);
   };
 
   return (
-    <Flex background="linear-gradient(180deg, #EAFFA9 24.48%, #EDEFFF 100%)" position="relative">
-      <Flex width={{ base: "100%", md: "60%" }} px="30px" py="30px" direction="column">
-        <Box maxW="200px">
-          <Image src="/assets/onboarding_batches_logo.png" alt="onboarding batches illustration: futuristic train" />
-        </Box>
-        <Text fontSize="30px" fontWeight="bold" lineHeight="1.2" color="gray.800">
-          Welcome to BuidlGuidl!
-        </Text>
-        <Text fontSize="sm" my="2" color="gray.800">
-          To help you get started in the Ethereum ecosystem, we've created the BuidlGuidl onboarding batch program. Dive
-          into end-to-end dApp development, receive mentorship from BG members, and learn how to collaborate with fellow
-          developers in open-source projects.
-        </Text>
-        <Spacer />
-        <Flex alignItems="center" flexDirection="row">
-          <Box mr="5">
-            {hasBatchNumber ? (
-              <Button
-                colorScheme="blue"
-                size="sm"
-                onClick={handleJoinTelegram}
-                isDisabled={!isInRecentBatch}
-                width="180px"
-              >
-                Join Telegram
-              </Button>
-            ) : (
-              <Button colorScheme="orange" size="sm" onClick={handleUpdateBatch} width="180px">
-                Get Telegram Access
-              </Button>
-            )}
+    <>
+      {!isLoadingOpenBatch && latestOpenBatch && (!hasBatchNumber || isInRecentBatch) && (
```

Comment:
A better to way to handle this: invert the condition and do an early return (easier to parse)

https://github.com/scaffold-eth/buidlguidl-v3/pull/117/commits/39b9e5542c0db55071d772476091311d6b941a1f